packages/pwa-kit-dev/src/ssr/server/build-dev-server.js

```diff
@@ -132,38 +133,92 @@ export const DevServerMixin = {
+            additionalProperties: {extensions: extensions}
+        })
+
         app.__extensions = extensions || []
 
         extensions.forEach((extension) => {
+            logger.info(`Loading extension: ${extension}`)
+
             const setupServerFilePathBase = path.join(
                 options.projectDir,
                 'node_modules',
                 extension,
+                'src',
                 'setup-server'
             )
-
             let filePath
             if (fs.existsSync(`${setupServerFilePathBase}.ts`)) {
                 filePath = `${setupServerFilePathBase}.ts`
             } else if (fs.existsSync(`${setupServerFilePathBase}.js`)) {
                 filePath = `${setupServerFilePathBase}.js`
             } else {
+                logger.warn(`No setup-server file found for ${extension}. Skipping.`)
                 return
             }
 
-            const setupServer = tsx.require(filePath, __filename)
-            if (!setupServer.default) {
-                console.warn(`Extension ${extension} does not have a default export. Skipping.`)
+            let ExtensionClass
+            try {
+                ExtensionClass = tsx.require(filePath, __filename).default
+            } catch (e) {
+                logger.error(`Error loading extension ${extension}:`, {
+                    namespace: 'DevServerMixin._setupExtensions',
+                    additionalProperties: {error: e}
+                })
                 return
             }
+
+            // Ensure that the default export is a class that implements IApplicationExtension
+            if (!ExtensionClass || typeof ExtensionClass !== 'function') {
+                logger.warn(`Extension ${extension} does not export a valid class. Skipping.`)
+                return
+            }
+
+            let extensionInstance
             try {
-                setupServer.default({app, options})
+                logger.info(`Instantiating extension class for ${extension}...`)
+                extensionInstance = new ExtensionClass(options)
+                logger.info(`Successfully instantiated extension ${extension}.`)
             } catch (e) {
-                console.error(`Error setting up extension ${extension}:`, e)
+                logger.error(`Error instantiating extension ${extension}:`, {
+                    namespace: 'DevServerMixin._setupExtensions',
+                    additionalProperties: {error: e}
+                })
+                return
+            }
+
+            // Verify the instance has the methods defined by IApplicationExtension
+            if (
+                typeof extensionInstance.getName !== 'function' ||
+                typeof extensionInstance.extendApp !== 'function'
+            ) {
+                logger.warn(
+                    `Extension ${extension} does not implement IApplicationExtension interface. Skipping.`
+                )
+                return
+            }
```

Comment:
In my latest changes in an open PR we'll be using an abstract class so we might be able to check if the class instance inherits from the base class in a more robust way.. like using the `instanceOf` operator.

Reply:
Let's syn on if we want to use an interface vs an abstract class and in which package we should define those class definitions.